editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/custom/GoIntoContainerFeature.java

```diff
@@ -58,24 +59,32 @@ public boolean canExecute(ICustomContext context) {
 			Object bo = getBusinessObjectForPictogramElement(pes[0]);
 			if (bo instanceof AbstractCamelModelElement) {
 				AbstractCamelModelElement cme = (AbstractCamelModelElement)bo;
-				
-				// go into is allowed if:
-				// - selected element is a route
-				// - selected container is the camel context
-				// - context contains more than one route
-				// go up is allowed if:
-				// - selected container is a route
-				// - selected container is the selected element
-				return 	(cme.getNodeTypeId().equals("route") && 
-						 CamelUtils.getDiagramEditor().getSelectedContainer() instanceof CamelContextElement &&
-						 cme.getCamelContext().getChildElements().size() > 1) ||
-						(CamelUtils.getDiagramEditor().getSelectedContainer().equals(cme) && 
-						 cme instanceof CamelRouteElement);
+				return isAllowedToExecute(cme);
 			}
 		}
 		return ret;
 	}
 	
+	private boolean isAllowedToExecute(AbstractCamelModelElement cme) {
+		// go into is allowed if:
+		// - selected element is a route
+		// - selected container is the camel context
+		// - context contains more than one route
+		// go up is allowed if:
+		// - selected container is a route
+		// - selected container is the selected element
+		CamelDesignEditor editor = (CamelDesignEditor)getDiagramBehavior().getDiagramContainer();
+		return 	(editor != null && 
+				 cme.getNodeTypeId().equals("route") && 
```

Comment:
maybe we can extract each side of the ||
if I understand well th ecomment, we will have something like isGoIntoAllowedToExecute(editor, cme)|| isGoUpAllowedToExecute(editor, cme)
The condition will easier to understand, what do you think?